### PR TITLE
Update the bug template to include commit ID

### PR DESCRIPTION
**Pull request by junbo75**
Signed-off-by: Junbo Liang <68558268+junbo75@users.noreply.github.com>

## What does this PR do?

Add the commit ID information to the GHI bug template which helps people to check whether the bug has been fixed in the latest for proper triage.

## How was this PR tested?

Cannot test the change locally. Will try to create a new GHI after the merge to verify it.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_template.md [hunk 1]
```diff
@@ -32,6 +32,9 @@ If applicable, add screenshots and/or a video to help explain your problem.
 **Found in Branch**
 Name of or link to the branch where the issue occurs.
 
+**Commit ID from Branch**
+ID (SHA-1 hash) of the latest commit from the branch where the issue occurs.
```
Comment (lmbr-pip) [inline .github/ISSUE_TEMPLATE/bug_template.md:36]: I would rephrase this for non-technical folks so it clear how to get the commit id and that commit ids only make sense if they are O3DE/O3DE commit ids.

Consider adding information about how to use `git rev-parse HEAD`or similar to get commit id

Comment (lmbr-pip) [reply .github/ISSUE_TEMPLATE/bug_template.md:36]: For example, GoDot's bug report language is "Specify the Git commit hash if using a development or non-official build. If you use a custom build, please test if your issue is reproducible in official builds too."

**Review by AMZN-stankowi — APPROVED** [review]
I'm glad you added the command to run to get the info, that will help a ton.

**Review by amzn-mike — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]
